controllers/vsl.go

```diff
@@ -19,61 +30,91 @@ func (r *VeleroReconciler) ValidateVolumeSnapshotLocations(log logr.Logger) (boo
 	// TODO: For each VSL, confirm for each selected provider, we have the
 	// needed config values
 
-	/*
-		for _, vslSpec := range velero.Spec.VolumeSnapshotLocations {
+	for i, vslSpec := range velero.Spec.VolumeSnapshotLocations {
+		vsl := velerov1.VolumeSnapshotLocation{
+			ObjectMeta: metav1.ObjectMeta{
+				// TODO: Use a hash instead of i
+				Name:      fmt.Sprintf("%s-%d", r.NamespacedName.Name, i+1),
+				Namespace: r.NamespacedName.Namespace,
+			},
+			Spec: vslSpec,
+		}
 
-			//AWS
-			if vslSpec.Provider == "aws" {
+		// check for valid provider
+		if vslSpec.Provider != AWSProvider && vslSpec.Provider != GCPProvider &&
+			vslSpec.Provider != Azure {
+			return false, errors.New("invalid provider for VSL")
```

Comment:
I'm wondering if a user configures a custom VSL plugin, we may be preventing them from adding it with this logic.

I think it would be better if we warned the user here instead. Maybe we trigger an event? Or warn the user on the status field somehow.

Reply:
Actually looking below we fire an event for something similar here: https://github.com/openshift/oadp-operator/pull/278/files#diff-9d604b6817079fa43c57dd7d9eaa35f87b4a609d7dae607aa4b64ca00e3d2cf9R66

Lets do that